lib/Interfaces/Activity/FollowInterface.php

```diff
@@ -0,0 +1,222 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+namespace OCA\Social\Interfaces\Activity;
+
+
+use daita\MySmallPhpTools\Exceptions\MalformedArrayException;
+use Exception;
+use OCA\Social\Db\FollowsRequest;
+use OCA\Social\Exceptions\FollowDoesNotExistException;
+use OCA\Social\Exceptions\InvalidOriginException;
+use OCA\Social\Exceptions\InvalidResourceException;
+use OCA\Social\Exceptions\ItemNotFoundException;
+use OCA\Social\Exceptions\RedundancyLimitException;
+use OCA\Social\Exceptions\Request410Exception;
+use OCA\Social\Exceptions\RequestException;
+use OCA\Social\Exceptions\SocialAppConfigException;
+use OCA\Social\Exceptions\UnknownItemException;
+use OCA\Social\Interfaces\IActivityPubInterface;
+use OCA\Social\Model\ActivityPub\ACore;
+use OCA\Social\Model\ActivityPub\Activity\Accept;
+use OCA\Social\Model\ActivityPub\Activity\Follow;
+use OCA\Social\Model\ActivityPub\Activity\Reject;
+use OCA\Social\Model\ActivityPub\Activity\Undo;
+use OCA\Social\Model\InstancePath;
+use OCA\Social\Service\ActivityService;
+use OCA\Social\Service\CacheActorService;
+use OCA\Social\Service\ConfigService;
+use OCA\Social\Service\MiscService;
+
+
+class FollowInterface implements IActivityPubInterface {
+
+
+	/** @var FollowsRequest */
+	private $followsRequest;
+
+	/** @var CacheActorService */
+	private $cacheActorService;
+
+	/** @var ActivityService */
+	private $activityService;
+
+	/** @var ConfigService */
+	private $configService;
+
+	/** @var MiscService */
+	private $miscService;
+
+
+	/**
+	 * NoteInterface constructor.
+	 *
+	 * @param FollowsRequest $followsRequest
+	 * @param CacheActorService $cacheActorService
+	 * @param ActivityService $activityService
+	 * @param ConfigService $configService
+	 * @param MiscService $miscService
+	 */
+	public function __construct(
+		FollowsRequest $followsRequest, CacheActorService $cacheActorService,
+		ActivityService $activityService, ConfigService $configService, MiscService $miscService
+	) {
+		$this->followsRequest = $followsRequest;
+		$this->cacheActorService = $cacheActorService;
+		$this->activityService = $activityService;
+		$this->configService = $configService;
+		$this->miscService = $miscService;
+	}
+
+
+	/**
+	 * @param ACore $item
+	 */
+	public function processResult(ACore $item) {
+	}
+
+
+	/**
+	 * @param Follow $follow
+	 */
+	public function confirmFollowRequest(Follow $follow) {
+		try {
+			$remoteActor = $this->cacheActorService->getFromId($follow->getActorId());
+
+			$accept = new Accept();
+			// TODO: improve the generation of the Id
+			$accept->setId($follow->getObjectId() . '#accepts/follows/' . rand(1000, 100000000));
```

Comment:
Please use generate() from ISecureRandom

Reply:
using the generateUniqueId() of the model that generate a uuid